dbt_semantic_interfaces/parsing/where_filter/query_interface.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from typing import Protocol, Sequence
+
+
```

Comment:
Given that this file solely contains protocols, what are people's thoughts on moving it the the `protocols` directory? cc: @plypaul & @tlento